src/catalog/catalog.cpp

```diff
@@ -1099,23 +1316,29 @@ void Catalog::AddBuiltinFunction(
  * @param   code_context  the code_context that holds the generated LLVM
  *                        query code
  */
-void Catalog::AddPlpgsqlFunction(
-    const std::string &name, const std::vector<type::TypeId> &argument_types,
-    const type::TypeId return_type, oid_t prolang, const std::string &func_src,
-    std::shared_ptr<peloton::codegen::CodeContext> code_context,
-    concurrency::TransactionContext *txn) {
+void Catalog::AddPlpgsqlFunction(concurrency::TransactionContext *txn,
```

Comment:
I don't think that this should be called `AddPlpgsqlFunction`. You are passing in the `prolang` argument, so it should just be called `AddFunction`, right? Furthermore, we are are referring to UDFs as procedures (i.e., `pg_proc` table), so it really should be called `AddProcedure`.

Reply:
I had no idea what this is supposed to do. Will change.